tools/topology/sof-tgl-rt711-rt1308.m4

```diff
@@ -189,18 +226,21 @@ DAI_CONFIG(ALH, 2, 0, SDW0-Playback,
 DAI_CONFIG(ALH, 3, 1, SDW0-Capture,
 	ALH_CONFIG(ALH_CONFIG_DATA(ALH, 3, 48000, 2)))
 
+ifdef(`EXT_AMP',
+`
 #ALH SDW1 Pin2 (ID: 2)
 DAI_CONFIG(ALH, 0x102, 2, SDW1-Playback,
 	ALH_CONFIG(ALH_CONFIG_DATA(ALH, 0x102, 48000, 2)))
+')
 
-# 3 HDMI/DP outputs (ID: 5,6,7)
-DAI_CONFIG(HDA, 0, 5, iDisp1,
+# 3 HDMI/DP outputs
+DAI_CONFIG(HDA, 0, HDMI_BELINK_ID, iDisp1,
 	HDA_CONFIG(HDA_CONFIG_DATA(HDA, 0, 48000, 2)))
-DAI_CONFIG(HDA, 1, 6, iDisp2,
+DAI_CONFIG(HDA, 1, eval(HDMI_BELINK_ID + 1), iDisp2,
 	HDA_CONFIG(HDA_CONFIG_DATA(HDA, 1, 48000, 2)))
-DAI_CONFIG(HDA, 2, 7, iDisp3,
+DAI_CONFIG(HDA, 2, eval(HDMI_BELINK_ID + 2), iDisp3,
 	HDA_CONFIG(HDA_CONFIG_DATA(HDA, 2, 48000, 2)))
-DAI_CONFIG(HDA, 3, 8, iDisp4,
+DAI_CONFIG(HDA, 3, eval(HDMI_BELINK_ID + 3), iDisp4,
 	HDA_CONFIG(HDA_CONFIG_DATA(HDA, 3, 48000, 2)))
```

Comment:
Given this is TGL+ tplg file, you could add the 4th PCM node for iDisp here. sof_sdw.c already has the support.

Reply:
Yes, I will do it.

Reply:
@kv2019i Oh, I almost forget my first design to use 3 hdmi. I use 3 hdmi pipeline intentionally here. This is because adl chrome platform (this tplg will be used by ADL chrome platform) DMI is not used by the driver, so the driver will only create 3 hdmi BE link and this will cause driver probe failure. This will be unmatched with the adl-p chrome platform until chrome's coreboot fixed the DMI informationl. Please see the linux driver at https://github.com/thesofproject/linux/blob/b7cd5a3826b69fef7ab813233c6583d5e774a9fb/sound/soc/intel/boards/sof_sdw.c#L193 . What about to use 4 HDMI pipelines after chrome's coreboot fixes the DMI issue?